src/Collision_TEST.cc

```diff
@@ -270,6 +279,60 @@ TEST(DOMCollision, CorrectBoxCollisionCalculateInertial)
   EXPECT_EQ(expectedInertial.Pose(), link->Inertial().Pose());
 }
 
+/////////////////////////////////////////////////
+TEST(DOMCollision, CalculateInertialWithAutoInertiaParamsElement)
+{
+  // This example considers a custom voxel-based inertia calculator
+  // <auto_inertial_params> element is used to provide properties
+  // for the custom calculator like voxel size, grid type, etc.
+  std::string sdf = "<?xml version=\"1.0\"?>"
+  " <sdf version=\"1.11\">"
+  "   <model name='shapes'>"
+  "     <link name='link'>"
+  "       <inertial auto='true' />"
+  "       <collision name='box_col'>"
+  "         <auto_inertia_params>"
+  "           <gz:voxel_size>0.01</gz:voxel_size>"
+  "           <gz:voxel_grid_type>float</gz:voxel_grid_type>"
+  "         </auto_inertia_params>"
+  "         <density>1240.0</density>"
+  "         <geometry>"
+  "           <box>"
+  "             <size>2 2 2</size>"
+  "           </box>"
+  "         </geometry>"
+  "       </collision>"
+  "     </link>"
+  "   </model>"
+  " </sdf>";
+
+  sdf::Root root;
+  const sdf::ParserConfig sdfParserConfig;
+  sdf::Errors errors = root.LoadSdfString(sdf, sdfParserConfig);
+  EXPECT_TRUE(errors.empty());
+  EXPECT_NE(nullptr, root.Element());
+
+  const sdf::Model *model = root.Model();
+  const sdf::Link *link = model->LinkByIndex(0);
+  const sdf::Collision *collision = link->CollisionByIndex(0);
+
+  root.ResolveAutoInertials(errors, sdfParserConfig);
+  EXPECT_TRUE(errors.empty());
+
+  sdf::ElementPtr autoInertiaParamsElem = collision->AutoInertiaParams();
+
+  // <auto_inertial_params> element is used as parent element for custom
+  // intertia calculator params. Custom elements have to be defined with a
+  // namespace prefix(gz in this case). More about this can be found in the
+  // following proposal:
+  // http://sdformat.org/tutorials?tut=custom_elements_attributes_proposal&cat=pose_semantics_docs&
```

Comment:
Can you also put this in the description of the `<auto_inertial_params>` in `collision.sdf` and `inertial.sdf`?

Reply:
I updated the description in [e921982](https://github.com/gazebosim/sdformat/pull/1304/commits/e921982bb49cff054209aaae5f5913aa2d661f6c)

Reply:
I don't see it in `collision.sdf`. Also, the but about custom elements and the link to the sdf proposal would be good to include.